### PR TITLE
New code excerpt project for cookbook/effects/download-button

### PR DESCRIPTION
**Pull request by khanhnwin**
_Description of what this PR is changing or adding, and why:_ More excerpts! Notice I commented out the ellipsis replacement in tool/refresh-code-excerpts.sh because it was accidentally replacing TextOverflow.ellipsis. 

Addresses the conversation in PR #6637 about moving this code to a new project to be analyzed.

_Issues fixed by this PR (if any):_

## Presubmit checklist
- [X] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [X] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [X] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Comment by sfshaza2** [comment]
I defer to @domesticmouse and @johnpryan. :D

**Comment by khanhnwin** [comment]
Forgot to refresh excerpts after fixing the excerpt script. Just refreshed the excerpts + build has kicked off again.

File: examples/cookbook/effects/download_button/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+include: package:flutter_lints/flutter.yaml
+
+linter:
+  rules:
+    prefer_single_quotes: true
```
Comment (domesticmouse) [inline examples/cookbook/effects/download_button/analysis_options.yaml:5]: `avoid_print`? =)

File: src/cookbook/effects/download-button.md [hunk 2]
```diff
@@ -76,13 +79,14 @@ class DownloadButton extends StatelessWidget {
...
+  final DownloadStatus status;
+  final Duration transitionDuration;
 
- Widget build(BuildContext context) {
-   // TODO:
-   return SizedBox();
- }
+  @override
+  Widget build(BuildContext context) {
+    // TODO:
```
Comment (domesticmouse) [inline src/cookbook/effects/download-button.md:87]: Please add a couple of words to the todo line =)

**Review by domesticmouse — APPROVED** [review]
LGTM

File: src/cookbook/effects/download-button.md (hunk 2 above)
Comment (khanhnwin) [reply src/cookbook/effects/download-button.md]: Done!

File: examples/cookbook/effects/download_button/analysis_options.yaml (hunk 1 above)
Comment (khanhnwin) [reply examples/cookbook/effects/download_button/analysis_options.yaml:6]: Done!

**Comment by khanhnwin** [comment]
Put in your suggestions. Build kicking off again :) Thanks @domesticmouse!!

**Pull request merged**